scipy/special/wright_bessel.pxd

```diff
@@ -0,0 +1,685 @@
+            0.04432750433880328, 0.04695505130394843, 0.04940093844946632,
+            0.05165570306958114, 0.05371062188899625, 0.05555774480621252,
+            0.05718992564772838, 0.05860084981322245, 0.05978505870426546,
+            0.06073797084177022, 0.06145589959031666, 0.06193606742068324,
+            0.06217661665534726, 0.06217661665534726, 0.06193606742068324,
+            0.06145589959031666, 0.06073797084177022, 0.05978505870426546,
+            0.05860084981322245, 0.05718992564772838, 0.05555774480621252,
+            0.05371062188899625, 0.05165570306958114, 0.04940093844946632,
+            0.04695505130394843, 0.04432750433880328, 0.0415284630901477,
+            0.03856875661258768, 0.03545983561514615, 0.03221372822357802,
+            0.0288429935805352, 0.02536067357001239, 0.02178024317012479,
+            0.01811556071348939, 0.01438082276148557, 0.01059054838365097,
+            0.006759799195745401, 0.002908622553155141]
+        # Fitted parameters for optimal choice of eps
+        # Call: python _precompute/wright_bessel.py 4
+        double *A = [0.41037, 0.30833, 6.9952, 18.382, -2.8566, 2.1122]
+
+    # We use the free choice of eps to make the integral better behaved.
+    # 1. Concern is int_0 K ~ int_0 (r+eps)^(-b) .. dr
+    #    This is difficult as r -> 0  for large b. It behaves better for larger
+    #    values of eps.
+    # 2. Concern is oscillatory behaviour of P. Therefore, we'd like to
+    #    make the change in the argument of cosine small, i.e. make arc length
+    #    int_0^phi sqrt(1 + f'(phi)^2) dphi small, with
+    #    f(phi) = eps * sin(phi) - x * eps^(-a) * sin(a*phi) + (1-b)*phi
+    #    Proxy, make |f'(phi)| small.
+    if b >= 8:
+        # Make P small compared to K by setting eps large enough.
+        # int K ~ exp(-eps) and int P ~ eps^(1-b)
+        eps = pow(b, -b / (1. - b))
+    else:
+        # Minimize oscillatory behaviour of P.
+        eps = (A[0] * b * exp(-0.5 * a)
+               + exp(A[1] + 1 / (1 + a) * log(x) - A[2] * exp(-A[3] * a)
+                     + A[4] / (1 + exp(A[5] * a))))
+        if a >= 4 and x >= 100:
+            eps += 1  # This part is hard to fit.
+    # safeguard, higher better for larger a, lower better for tiny a.
+    eps = min(eps, 100.)
+    eps = max(eps, 3.)  # Note: 3 seems to be a pretty good choice in general.
+
+    res1 = 0
+    res2 = 0
+    for k in range(50):
+        res1 += w_laguerre[k] * _Kmod(eps, a, b, x, x_laguerre[k])
+        # y = (b-a)*(x+1)/2.0 + a  for integration from a=0 to b=pi
+        y = M_PI * (x_legendre[k] + 1) / 2.0
+        res2 += w_legendre[k] * _P(eps, a, b, x, y)
+    res1 *= exp(-eps)
+    # (b-a)/2.0 * np.sum(w*func(y, *args), axis=-1)
+    res2 *= M_PI/2.0
+    res2 *= pow(eps, 1-b)
+
+    return 1./M_PI * (res1 + res2)
+
+
+@cython.cdivision(True)
+cdef inline double wright_bessel_scalar(double a, double b, double x) nogil:
```

Comment:
Is there a point where we know we're going to start losing accuracy (large `x`, `a`, `b`)? If so it would be good to start returning NaNs in that region instead of inaccurate results.

Reply:
`b >= rgamma_zero` returns nan, also some others overflow correctly. A hard region a<5, large b (10) and large x (>500), see also the `test_wright_data_grid_failures`.
So far, I've aimed for 1e-11 accuracy. After much work, this seems possible with a few exceptions.